pkg/kubelet/cm/cpumanager/policy_static_test.go

```diff
@@ -939,52 +939,77 @@ func TestTopologyAwareAllocateCPUs(t *testing.T) {
+			description:      "empty cpuset",
+			topo:             topoDualSocketHT,
+			numReservedCPUs:  2,
+			reserved:         cpuset.New(0, 1),
+			cpuPolicyOptions: map[string]string{},
+			stAssignments:    state.ContainerCPUAssignments{},
+			stDefaultCPUSet:  cpuset.New(),
+			expCSet:          cpuset.New(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11),
 		},
 		{
-			description:     "reserved cores 0 & 1 are not present in available cpuset",
-			topo:            topoDualSocketHT,
-			numReservedCPUs: 2,
-			reserved:        cpuset.New(0, 1),
-			stAssignments:   state.ContainerCPUAssignments{},
-			stDefaultCPUSet: cpuset.New(2, 3, 4, 5),
-			expErr:          fmt.Errorf("not all reserved cpus: \"0-1\" are present in defaultCpuSet: \"2-5\""),
+			description:      "empty cpuset with StrictCPUReservationOption enabled",
+			topo:             topoDualSocketHT,
+			numReservedCPUs:  2,
+			reserved:         cpuset.New(0, 1),
+			cpuPolicyOptions: map[string]string{StrictCPUReservationOption: "true"},
+			stAssignments:    state.ContainerCPUAssignments{},
+			stDefaultCPUSet:  cpuset.New(),
+			expCSet:          cpuset.New(2, 3, 4, 5, 6, 7, 8, 9, 10, 11),
 		},
 		{
-			description:     "inconsistency between numReservedCPUs and reserved",
-			topo:            topoDualSocketHT,
-			numReservedCPUs: 1,
-			reserved:        cpuset.New(0, 1),
-			stAssignments:   state.ContainerCPUAssignments{},
-			stDefaultCPUSet: cpuset.New(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11),
-			expNewErr:       fmt.Errorf("[cpumanager] unable to reserve the required amount of CPUs (size of 0-1 did not equal 1)"),
+			description:      "reserved cores 0 & 1 are not present in available cpuset",
+			topo:             topoDualSocketHT,
+			numReservedCPUs:  2,
+			reserved:         cpuset.New(0, 1),
+			cpuPolicyOptions: map[string]string{},
+			stAssignments:    state.ContainerCPUAssignments{},
+			stDefaultCPUSet:  cpuset.New(2, 3, 4, 5),
+			expErr:           fmt.Errorf("not all reserved cpus: \"0-1\" are present in defaultCpuSet: \"2-5\""),
+		},
+		{
+			description:      "reserved cores 0 & 1 are present in available cpuset with StrictCPUReservationOption enabled",
+			topo:             topoDualSocketHT,
+			numReservedCPUs:  2,
+			reserved:         cpuset.New(0, 1),
+			cpuPolicyOptions: map[string]string{StrictCPUReservationOption: "true"},
+			stAssignments:    state.ContainerCPUAssignments{},
+			stDefaultCPUSet:  cpuset.New(0, 1, 2, 3, 4, 5),
+			expErr:           fmt.Errorf("some of strictly reserved cpus: \"0-1\" are present in defaultCpuSet: \"0-5\""),
+		},
+		{
+			description:      "inconsistency between numReservedCPUs and reserved",
+			topo:             topoDualSocketHT,
+			numReservedCPUs:  1,
+			reserved:         cpuset.New(0, 1),
+			cpuPolicyOptions: map[string]string{},
+			stAssignments:    state.ContainerCPUAssignments{},
+			stDefaultCPUSet:  cpuset.New(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11),
```

Comment:
this highlights a beta blocker, for when we enable the feature by default.
when updating an older kubelet which almost certainly will have the reserved cpus in the default cpu set, the kubelet willl fail to start. The only clean approach atm seems to drain the node and remove the cpumanager state file.
/cc @jingczhang

Reply:
Hi @ffromani, removing the cpumanager state file before restarting kubelet is feature (de-)activation procedure, documented in the feature's KEP, documentation and blogs already. There is no need to drain the node, Kubelet is able to rebuild the runtime configuration after the restart, burstable and best-effort pods inherit kubelet's new cpuset (with reserved CPU cores removed or added back). Please review and check.

Reply:
I will review the upgrade notes. It's probably fine without the draining
1. if we are going from option disabled to enabled, we're being more restrictive, so it's probably fine
2. if we are going from enabled to disabled, we won't hit this path anyway

In this scenario the only (admittedly remote) concern is the shared pool exhaustion, but for that we will depend on the metrics to assess how severe this would be.

Thus it seems indeed we're all set.